src/ashley/views.py

```diff
@@ -118,6 +112,37 @@ def _do_on_login(self, lti_request: LTI) -> HttpResponse:
 
         return response
 
+    @classmethod
+    def _check_marked_locked_unlocked_unsync(cls, forum, context):
+        """
+        The context contains a field is_marked_locked, we control that if
+        this field is checked or unchecked, there is no problem of unsynchronicity
+        It's a security to remove/add forum permission on the default group.
+        This could only happen if the course was locked/unlocked outside
+        web navigation directly on the database
+        """
+        group_permissions = GroupForumPermission.objects.filter(
+            forum=forum, group=context.get_base_group(), has_perm=True
+        ).values_list("permission__codename", flat=True)
+        # forum is marked locked but the base group have all writing permissions
+        locked_unsync = context.is_marked_locked and any(
+            item in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS for item in group_permissions
+        )
+
+        # forum is marked unlocked but the base group don't have all the writing
+        # permissions
+        unlocked_unsync = not context.is_marked_locked and not all(
+            item in group_permissions for item in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS
+        )
+
+        if locked_unsync or unlocked_unsync:
+            for perm in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS:
+                # pylint: disable=expression-not-assigned
+                if locked_unsync:
+                    remove_perm(perm, context.get_base_group(), forum)
+                elif perm not in group_permissions:
+                    assign_perm(perm, context.get_base_group(), forum)
```

Comment:
Maybe clearer like this?
```suggestion
        if context.is_marked_locked:
            for item in group_permissions:
                if item in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS:
                    # forum is marked locked but the base group still has a writing permission
                     remove_perm(perm, context.get_base_group(), forum)
        else:
            for item in DEFAULT_FORUM_BASE_WRITE_PERMISSIONS:
                if item not in group_permissions:
                    # forum is marked unlocked but the base group doesn't have all the writing permissions
                    assign_perm(perm, context.get_base_group(), forum)
```